packages/cli/src/utils/generate.ts

```diff
@@ -286,6 +286,46 @@ function updateBuildTime(basePath: string) {
   }
 }
 
+function checkDependencies(basePath: string, packagesToCheck: string[]) {
+  const { coreDir, getRoot } = withBasePath(basePath)
+
+  const corePackageJsonPath = path.join(coreDir, 'package.json')
+  const rootPackageJsonPath = path.join(getRoot(), 'package.json')
+
+  const corePackageJson = require(corePackageJsonPath)
+  const rootPackageJson = require(rootPackageJsonPath)
+
+  packagesToCheck.forEach((packageName) => {
+    const coreVersion =
+      corePackageJson.devDependencies[packageName] ||
+      corePackageJson.dependencies[packageName]
+    const rootVersion =
+      rootPackageJson.devDependencies[packageName] ||
+      rootPackageJson.dependencies[packageName]
+
+    if (!coreVersion || !rootVersion) {
+      console.warn(
+        `${chalk.yellow(
+          'warning'
+        )} - Package ${packageName} not found in both core or root dependencies.`
+      )
+    } else if (coreVersion !== rootVersion) {
```

Comment:
If you compare the versions this way, it will break all the stores that update the **Faststore** version. Think about the case where the store has a customization that uses a new typescript functionality. The dev will have to decrement the typescript version to be able to build the project and will have to change its implementation by removing the ts feature that does not exist in previous versions.

Reply:
But we don't agree in this case, we want the store to use the same version of typescript that we use in order to avoid nonsupport on build.

Reply:
For a v1 you can show a warning to the developer. This will make it easier for him and us to track build errors related to typescript. For a v2, you can make more specific comparisons by looking at patch, minor and major.

Reply:
This is only to v3.

Reply:
So hahaha. As there are few stores in v3. We can break the build and communicate to the stores to update their version. Or we can fail the build only on major diffs and throw a warning for minor and patch diffs